src/object/cli_ec.c

```diff
@@ -2567,3 +2567,68 @@ obj_ec_tgt_oiod_init(struct obj_io_desc *r_oiods, uint32_t iod_nr,
 
 	return tgt_oiods;
 }
+
+/* Get all of recxs of the specific target from the daos offset */
+int
+obj_recx_ec2_daos(struct daos_oclass_attr *oca, int shard,
+		  daos_recx_t *recxs, int nr, daos_recx_t **output_recxs,
+		  int *output_nr)
+{
+	daos_recx_t *tgt_recxs;
+	int cell_bytes = obj_ec_cell_rec_nr(oca);
+	int stripe_bytes = obj_ec_stripe_rec_nr(oca);
+	int tgt_idx;
+	int total = 0;
+	int idx = 0;
+	int i;
+
+	if (oca->ca_resil == DAOS_RES_REPL)
+		return 0;
+
+	tgt_idx = shard % obj_ec_tgt_nr(oca);
+	/* parity shard conversion */
+	if (tgt_idx >= obj_ec_data_tgt_nr(oca)) {
+		for (i = 0; i < nr; i++) {
+			daos_off_t offset = recxs[i].rx_idx;
+
+			if (!(offset & PARITY_INDICATOR))
+				continue;
+			recxs[i].rx_idx = obj_ec_parity_off_to_daos_off(offset,
+						cell_bytes, stripe_bytes);
+			recxs[i].rx_nr *= obj_ec_data_tgt_nr(oca);
+		}
+		*output_recxs = recxs;
+		*output_nr = nr;
+		return 0;
+	}
+
+	/* data shard conversion */
+	*output_nr = 0;
+	for (i = 0; i < nr; i++)
+		total += (recxs[i].rx_nr + cell_bytes - 1) / cell_bytes;
```

Comment:
oh, this recxs is VOS extents, instead of DAOS extents. It looks like obj_ec_recx_cell_nr is for daos extent? though I probably need change name here.